### PR TITLE
fix(fixed_point): snap pow base near 1.0 to dodge log2 polynomial residual

### LINKED ISSUE #2969: fl::fixed_point::pow boundary handling: x=0 and x≈1 produce wrong results
## Summary
`fl::s8x24::pow` (and other `fl::fixed_point<I,F>::pow` specializations) give wrong / lossy results at the input boundaries `x = 0` and `x ≈ 1`. Callers currently have to special-case these manually.

## Repro
Found while building `Gamma8Impl` LUT (PR #2968, `src/fl/math/ease.cpp.hpp`):

```cpp
const fl::s8x24 gamma_fp(3.2f);
constexpr fl::s8x24 inv_255_fp(1.0f / 255.0f);

// i = 0: x = 0 — pow should return 0 (mathematically pow(0, any) = 0
// for positive exponents). Implementation returns 0 OK but only because
// of an early-out check in the wrapper, not because the algorithm
// handles it.

// i = 255: x = 255 * (1/255) — but 1/255 in s8x24 is not exact, so
// x lands at raw 0x00FFFFFF = 0.99999994 (one LSB below 1.0).
const fl::s8x24 x   = static_cast<i32>(255) * inv_255_fp;
const fl::s8x24 r   = fl::s8x24::pow(x, gamma_fp);
// r ≈ 0.99902 instead of expected 1.0 — off by ~57 LSB when scaled to u16
// (returns 65478 instead of 65535).
```

## Root cause
`pow(base, exp)` is implemented as `exp2_fp(exp * log2_fp(base))`. The `log2_fp` helper uses a 4-term minimax polynomial for `log2(1+t)` on `t ∈ [0, 1)`. At the upper endpoint `t→1`:

- True value: `log2(2) = 1.0`
- Polynomial: `c0 + c1 + c2 + c3 = 0.999557` — off by 0.000443

When the input is one LSB below 1.0, the polynomial path is taken instead of an exact-1.0 short-circuit, and the 0.000443 error propagates through `exp2_fp`, then scales up to ~50-100 LSB when mapped to u16.

## Suggested fix
Add explicit short-circuits in `pow` BEFORE invoking `exp2_fp(exp * log2_fp(...))`:

```cpp
static FASTLED_FORCE_INLINE Derived pow(Derived base, Derived exp) FL_NOEXCEPT {
    if (base.mValue <= 0) return Derived();           // already present
    constexpr Derived one(1.0f);
    if (exp.mValue == 0) return one;                  // already present
    if (base == one) return one;                      // already present
    // NEW: snap base values within 1 ULP of 1.0 to exactly 1.0 to
    // dodge the log2(1+t) polynomial's upper-endpoint residual.
    if (base.mValue >= Derived::SCALE - SOME_TOLERANCE && base.mValue <= Derived::SCALE) {
        return one;
    }
    return exp2_fp(exp * log2_fp(base));
}
```

Or more rigorously: extend the `log2(1+t)` polynomial fit to be exact at `t=1` (one-sided constraint in the minimax solver).

## Affected types
All `fl::fixed_point<I,F>::pow` specializations that route through `exp2_fp(exp * log2_fp(base))`:
- `s4x12`, `s8x8`, `s8x24`, `s12x4`, `s16x16`, `s24x8`
- `u4x12`, `u8x8`, `u8x24`, `u12x4`, `u16x16`, `u24x8`

## Workaround (current)
`Gamma8Impl` in `src/fl/math/ease.cpp.hpp` manually sets `mLut[0] = 0` and `mLut[255] = 65535` to bypass the boundary issue. Should be removable once `pow` is fixed.

## Refs
- PR #2968 (where the workaround was added)
- #2963 (gamma audit context)

### PR DESCRIPTION
**Pull request by zackees**
Closes #2969

## Root cause

`fl::fixed_point<I,F>::pow(base, exp)` is implemented as `exp2_fp(exp * log2_fp(base))`. The `log2_fp` helper uses a 4-term minimax polynomial for `log2(1+t)` on `t ∈ [0, 1)`. At the upper endpoint `t → 1` the polynomial evaluates to ≈0.999557 instead of 1.0 — a 0.000443 residual that, after `exp2` and scale-to-u16, drops the output by ~50-100 LSB below the expected value.

The existing short-circuits caught `base.mValue == SCALE` exactly, `base.mValue <= 0`, and `exp.mValue == 0`. But values like `255 * (1/255)` in fixed-point land one ULP below 1.0 because `1/255` is not representable exactly — so the input never hit the early-out and went through the lossy polynomial path:

```cpp
pow(0.99999994, 3.2) // → ~0.99902 (raw 65478) instead of 1.0 (raw 65535)
```

## Fix

Add a near-1.0 snap before the polynomial dispatch in every `pow`: if `base.mValue` is within 2 ULPs of `SCALE`, return one directly. Applied uniformly across:

- `src/fl/math/fixed_point/base.h` (the templated base-class pow)
- 12 concrete specializations: `s4x12`, `s8x8`, `s8x24`, `s12x4`, `s16x16`, `s24x8`, `u4x12`, `u8x8`, `u8x24`, `u12x4`, `u16x16`, `u24x8`

## Bonus cleanup

Removed the `Gamma8Impl::mLut[255] = 65535` workaround that was added in #2968 to dodge this bug. The constructor now loops through `[1, 256)` and lets `s8x24::pow` produce the correct boundary value on its own. The explanatory comment block for the workaround is gone; `mLut[0] = 0` is kept (slightly cheaper than a loop branch and ensures mLut is initialised).

## Tests

Added 4 new `FL_TEST_CASE` groups in `tests/fl/math/fixed_point.cpp` (4 type subcases each — `s16x16`, `s8x24`, `u16x16`, `u8x24`):

- `pow boundary - exactly 1.0` — base == 1.0 still returns 1.0
- `pow boundary - one ULP below 1.0 snaps to 1.0` — the regression case
- `pow boundary - two ULPs below 1.0 snaps to 1.0` — upper edge of snap window
- `pow boundary - far from 1.0 not snapped` — `pow(0.5, 2)` ≈ 0.25 (signed) / `pow(2, 0.5)` ≈ 1.414 (unsigned) confirm the snap doesn't over-trigger

### Verified locally
- `bash test fl_math_fixed_point --cpp` — passes (including 16 new boundary subcases)
- `bash test fl_chipsets_ucs7604 --cpp` — passes (originally-failing gamma=3.2 at i=255 → 65535 case)
- `bash test fl_math_ease --cpp` — passes (Gamma8Impl rewrite)
- `bash lint --cpp` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved numerical precision by snapping fixed-point pow inputs very near 1.0 to exactly 1.0, avoiding tiny endpoint errors.
  * Refined gamma lookup-table generation to ensure correct endpoint handling.

* **Tests**
  * Added regression tests for pow boundary behavior.
  * Updated test baseline data to reflect new measurement.

### CODE REVIEW

File: tests/fl/math/fixed_point.cpp [hunk 1]
```diff
@@ -5125,6 +5125,103 @@ FL_TEST_CASE("Integer constructor - fixed_point<> wrapper delegation") {
...
+//
+// Background: `pow(base, exp)` is implemented as `exp2_fp(exp * log2_fp(base))`.
+// `log2_fp` uses a 4-term minimax polynomial for `log2(1+t)` on `t ∈ [0, 1)`.
+// At the upper endpoint t→1 the polynomial evaluates to ≈0.999557 instead of
+// 1.0 — a 0.000443 error that, after exp2 + scale-to-u16, drops the output
+// by ~50-100 LSB below the expected 65535. The original failure mode was
+// `Gamma8Impl::mLut[255]` returning 65478 instead of 65535 for gamma=3.2.
+//
+// Fix: snap `base.mValue` values within 2 ULPs of exactly-1.0 to one before
+// calling the polynomial. The tests below pin that contract.
```
[the remark this answers is not in the document]
Comment (zackees) [reply tests/fl/math/fixed_point.cpp]: Fixed in d5de870 -- comment block on lines 5132-5142 now uses 7-bit ASCII (`in [0, 1)`, `t -> 1`, `approx`, `--`).

File: tests/fl/math/fixed_point.cpp [hunk 2]
```diff
@@ -5125,6 +5125,103 @@ FL_TEST_CASE("Integer constructor - fixed_point<> wrapper delegation") {
...
+}
+
+// Same as above, two LSBs below 1.0 — also inside the snap window.
+template <typename FP>
+static void test_pow_two_ulps_below_one_impl() {
+    const FP just_below = FP::from_raw(static_cast<raw_t<FP>>(FP::SCALE - 2));
+    const FP one(1.0f);
+    const FP gamma_2_8(2.8f);
+    FL_CHECK_EQ(FP::pow(just_below, gamma_2_8).raw(), one.raw());
+}
```
[the remark this answers is not in the document]
Comment (zackees) [reply tests/fl/math/fixed_point.cpp:5178]: Fixed in d5de870 -- test_pow_two_ulps_below_one_impl now also asserts the gamma_3_2 case, mirroring the gamma fanout of test_pow_one_ulp_below_one_impl.
